satpy/dataset/data_dict.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+from .dataid import DataID, create_filtered_query, minimal_default_keys_config
+
+
+class TooManyResults(KeyError):
+    """Special exception when one key maps to multiple items in the container."""
+
+    pass
+
+
+def filter_keys_by_dataset_query(dquery, key_container):
+    """Filer provided key iterable by the provided `DataQuery`.
+
+    Note: The `modifiers` attribute of `did` should be `None` to allow for
+          **any** modifier in the results.
+
+    Args:
+        dquery (DataQuery): Query parameters to match in the `key_container`.
+        key_container (iterable): Set, list, tuple, or dict of `DataID`
+                                  keys.
+
+    Returns (list): List of keys matching the provided parameters in no
+                    specific order.
+
+    """
+    return dquery.filter_dataids(key_container)
```

Comment:
Maybe this function could be inlined altogether?

Reply:
Filter keys by parameter id was already there, I just replaced the content of it with a method call on the data query.